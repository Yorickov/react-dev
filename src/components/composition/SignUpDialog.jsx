import React, { Component } from 'react';
import Dialog from './Dialog.jsx';

class SignUpDialog extends Component {
  state = { login: '' };

  handleChange = ({ target: { value } }) => {
    this.setState({ login: value });
  };

  handleSignUp = () => {
    const { login } = this.state;
    alert(`Welcome to the board, ${login}!`); // eslint-disable-line
  };

  render() {
    const { login } = this.state;
    return (
      <Dialog
        title="Marse research program"
        message="What is your name?"
      >
        <input
          value={login}
          onChange={this.handleChange}
          className='from-control'
        />
        <button
          type="button"
          className="btn btn-primary"
          onClick={this.handleSignUp}
        >
          Take me!
        </button>
      </Dialog>
    );
  }
}

export default SignUpDialog;
