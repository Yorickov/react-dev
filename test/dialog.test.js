import React from 'react';
import SignUpDialog from '../src/components/composition/SignUpDialog.jsx';

it('SignUpDialog', () => {
  jest.spyOn(window, 'alert').mockImplementation(() => {});

  const wrapper = mount(<SignUpDialog />);
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('input').simulate('change', { target: { value: 'Pete' } });
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('button').simulate('click');
  expect(window.alert).toBeCalledWith('Welcome to the board, Pete!');
});
