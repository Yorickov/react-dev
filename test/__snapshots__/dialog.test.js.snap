// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpDialog 1`] = `
<div
  class="card text-primary"
>
  <div
    class="card-body"
  >
    <h2
      class="card-title"
    >
      Marse research program
    </h2>
    <p
      class="card-text"
    >
      What is your name?
    </p>
    <form>
      <input
        class="from-control"
        value=""
      />
      <button
        class="btn btn-primary"
        type="button"
      >
        Take me!
      </button>
    </form>
  </div>
</div>
`;

exports[`SignUpDialog 2`] = `
<div
  class="card text-primary"
>
  <div
    class="card-body"
  >
    <h2
      class="card-title"
    >
      Marse research program
    </h2>
    <p
      class="card-text"
    >
      What is your name?
    </p>
    <form>
      <input
        class="from-control"
        value="Pete"
      />
      <button
        class="btn btn-primary"
        type="button"
      >
        Take me!
      </button>
    </form>
  </div>
</div>
`;
